django/middleware/http.py

```diff
@@ -15,6 +18,9 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        if settings.USE_ETAGS and not response.has_header('ETag'):
```

Comment:
@MarkusH nope, only the setting check. 

This will be the place where the ETag will be generated.